kotlin/internal-testing-utils/src/main/java/com/squareup/workflow/internal/util/UncaughtExceptionGuard.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.internal.util
+
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * Helper for collecting uncaught exceptions and reporting them during the execution of some
+ * function.
+ *
+ * ## Usage
+ *
+ * 1. Register an uncaught exception handler to report uncaught exceptions with [reportUncaught].
+ * 2. Run your code with [runRethrowingUncaught]. Any exceptions passed to [reportUncaught] will be
```

Comment:
No discussion of `rethrowingUncaughtExceptions()`, so I come away unclear when I should use that and when I should use these methods directly.

Reply:
Added a note saying when to use it.